repobee_sanitizer/_sanitize_repo.py

```diff
@@ -34,6 +34,14 @@ def _sanitize_repo(
         if not args.file_list.is_file():
             raise plug.PlugError(f"No such file: {args.file_list}")
 
+        repo = git.Repo.init(args.repo_root)
+        if repo.head.commit.diff():
+            return plug.Result(
+                name="sanitize-repo",
+                msg="There are uncommited staged files in the repo",
+                status=plug.Status.ERROR,
+            )
+
```

Comment:
Make this more modular as discussed, since there will be two more almost identical cases to check.